File: mobile/testopithecus/PAL/DefaultReportIntegration.swift
//
//  DefaultReportIntegration.swift
//  YandexMobileMailAutoTests
//
//  Created by Artem Zoshchuk on 14.10.2020.
//  Copyright © 2020 Yandex. All rights reserved.
//

import Foundation
import testopithecus

public class DefaultReportIntegration: ReportIntegration {
    public func addFeatureName(_ feature: String) {
        Allure.setTestFeature(feature: feature)
    }
        
    public func addTestpalmId(_ id: Int32) {
        Allure.setTestId(testPalmCaseID: Int(id))
    }
}

File: mobile/testopithecus/DefaultReportIntegration.kt
package com.yandex.mail.testopithecus

import com.yandex.xplat.testopithecus.common.ReportIntegration
import io.qameta.allure.kotlin.Allure

class DefaultReportIntegration : ReportIntegration {
    override fun addTestpalmId(id: Int) {
        Allure.tms("$id", "https://testpalm.yandex-team.ru/testcase/mobilemail-$id")
    }

    override fun addFeatureName(feature: String) {
        Allure.feature(feature)
    }
}
